src/org/thoughtcrime/securesms/ProfileDocumentsFragment.java

```diff
@@ -172,39 +165,25 @@ public void onMediaLongClicked(DcMsg mediaRecord) {
     }
   }
 
-  @SuppressLint("StaticFieldLeak")
-  private void handleDeleteMedia(@NonNull Collection<DcMsg> mediaRecords) {
-    int recordCount       = mediaRecords.size();
-    Resources res         = getContext().getResources();
-    String confirmMessage = res.getQuantityString(R.plurals.ask_delete_messages,
-        recordCount,
-        recordCount);
-
-    AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
-    builder.setMessage(confirmMessage);
-    builder.setCancelable(true);
-
-    builder.setPositiveButton(R.string.delete, (dialogInterface, i) -> {
-      new ProgressDialogAsyncTask<DcMsg, Void, Void>(getContext(),
-          R.string.one_moment,
-          R.string.one_moment)
-      {
-        @Override
-        protected Void doInBackground(DcMsg... records) {
-          if (records == null || records.length == 0) {
-            return null;
-          }
-
-          for (DcMsg record : records) {
-            dcContext.deleteMsgs(new int[]{record.getId()});
-          }
-          return null;
-        }
-
-      }.execute(mediaRecords.toArray(new DcMsg[mediaRecords.size()]));
-    });
-    builder.setNegativeButton(android.R.string.cancel, null);
-    builder.show();
+  private DcMsg getSelectedMessageRecord() {
+    Set<DcMsg> messageRecords = getListAdapter().getSelectedMedia();
+
+    if (messageRecords.size() == 1) return messageRecords.iterator().next();
+    else                            throw new AssertionError();
+  }
+
+  private void setCorrectMenuVisibility(Menu menu) {
```

Comment:
This could go into MessageSelectorFragment too I guess, and then either be overridden in ConversationFragment? It's probably not possible to use the setCorrectMenuVisibility() from ConversationFragment for all three classes?

Reply:
yes, it could be added as an abstract method, didn't added it because it can't be reused by the three